src/main/java/br/com/caelum/vraptor/test/container/CdiContainer.java

```diff
@@ -13,12 +13,32 @@
         private Weld weld;
         private WeldContainer weldContainer;
         private Contexts contexts;
+    	
+    	public CdiContainer(Weld weld, WeldContainer weldContainer, Contexts contexts) {
+    		this(weld, weldContainer);
+    		this.contexts = contexts;
+    	}
 
-        public void start() {
-                weld = new Weld();
-                weldContainer = weld.initialize();
-                contexts = getContexts();
-        }
+    	public CdiContainer(Weld weld, WeldContainer weldContainer) {
+    		super();
+    		this.weld = weld;
+    		this.weldContainer = weldContainer;
```

Comment:
i'm just unsure about allowing user to provide a non-related instance of weld and its container. 
is there a case of use in which the weldcontainer isn't the weld#initialize output?